model-optimizer/mo/utils/ir_reader/extender.py

```diff
@@ -36,6 +37,17 @@ def attr_to_list(node: Node, attribute: str):
 
     @staticmethod
     def const_shape_infer(node: Node):
+        # Check equality of old (restored from IR) and new (calculated while shape inference) input shapes
+        node['new_input_shapes'] = list()
+        for n in node.in_nodes():
+            node.new_input_shapes.append(node.in_node(n).shape)
+        assert len(node.new_input_shapes) == len(node.old_input_shapes), \
+            'Something wrong happened while {} node with type {} copy shape inference!'.format(node.name, node.type)
+        for i in range(len(node.new_input_shapes)):
+            assert np.array_equal(node.new_input_shapes[i], node.old_input_shapes[i]), \
+                'Something wrong happened while {} node with type {} copy shape inference!'.format(node.name, node.type)
+
+        # Set all output shapes the same as restored from IR
```

Comment:
I think that it can be re-written to new Graph API. There is small amount of code:)

Reply:
Yes, done.